lib/gl_rubocop/gl_cops/tailwind_no_contradicting_class_name.rb

```diff
@@ -0,0 +1,243 @@
+        # Display
+        'block' => %w[block hidden flex inline inline-block inline-flex grid inline-grid table],
+        'hidden' => %w[block hidden flex inline inline-block inline-flex grid inline-grid table],
+        'flex' => %w[block hidden flex inline inline-block inline-flex grid inline-grid table],
+        'inline' => %w[block hidden flex inline inline-block inline-flex grid inline-grid table],
+        'inline-block' => %w[block hidden flex inline inline-block inline-flex grid inline-grid
+                             table],
+        'inline-flex' => %w[block hidden flex inline inline-block inline-flex grid inline-grid
+                            table],
+        'grid' => %w[block hidden flex inline inline-block inline-flex grid inline-grid table],
+        'inline-grid' => %w[block hidden flex inline inline-block inline-flex grid inline-grid
+                            table],
+        'table' => %w[block hidden flex inline inline-block inline-flex grid inline-grid table],
+        # Position
+        'absolute' => %w[static relative absolute fixed sticky],
+        'relative' => %w[static relative absolute fixed sticky],
+        'fixed' => %w[static relative absolute fixed sticky],
+        'static' => %w[static relative absolute fixed sticky],
+        'sticky' => %w[static relative absolute fixed sticky],
+        # Text alignment
+        'text-left' => %w[text-left text-center text-right text-justify],
+        'text-center' => %w[text-left text-center text-right text-justify],
+        'text-right' => %w[text-left text-center text-right text-justify],
+        'text-justify' => %w[text-left text-center text-right text-justify],
+        # Flex direction
+        'flex-row' => %w[flex-row flex-row-reverse flex-col flex-col-reverse],
+        'flex-row-reverse' => %w[flex-row flex-row-reverse flex-col flex-col-reverse],
+        'flex-col' => %w[flex-row flex-row-reverse flex-col flex-col-reverse],
+        'flex-col-reverse' => %w[flex-row flex-row-reverse flex-col flex-col-reverse],
+        # Justify content
+        'justify-start' => %w[justify-start justify-end justify-center justify-between
+                              justify-around justify-evenly],
+        'justify-end' => %w[justify-start justify-end justify-center justify-between justify-around
+                            justify-evenly],
+        'justify-center' => %w[justify-start justify-end justify-center justify-between
+                               justify-around justify-evenly],
+        'justify-between' => %w[justify-start justify-end justify-center justify-between
+                                justify-around justify-evenly],
+        'justify-around' => %w[justify-start justify-end justify-center justify-between
+                               justify-around justify-evenly],
+        'justify-evenly' => %w[justify-start justify-end justify-center justify-between
+                               justify-around justify-evenly]
+      }.freeze
```

Comment:
I chose not to add anything related to colors here in the interest of keeping the rule less complex, as I didn't want to have to implement logic around reading our Tailwind config to generate the color-based classes and then match against them. Figure if we need to, we can always follow up on that